version: '3.7'

services:
  postgres:
    container_name: postgres
    image: postgres:14.3
    restart: always
    environment:
      - POSTGRES_USER=user_default
      - POSTGRES_PASSWORD=defaultdatabase
      - POSTGRES_DB=vl_database
    ports:
      - "5432:5432"
    volumes:
      - ./volumes/postgres_data:/var/lib/postgresql/data/
    networks:
      - backend

  django:
    build:
      context: .
    ports:
      - "8000:8000"
    volumes:
      - ./django:/vertical_logistics
    command: >
      sh -c "
        python manage.py collectstatic --noinput &&
        python manage.py migrate &&
        python manage.py test --failfast logistics authentication &&
        coverage run --source='./logistics' manage.py test &&
        coverage report &&
        coverage html &&
        python manage.py runserver 0.0.0.0:8000
      "
    user: root
    environment:
      - SECRET_KEY=1234
      - DEBUG=True
      - DB_NAME=vl_database
      - DB_USER=user_default
      - DB_PASSWORD=defaultdatabase
      - DB_ENGINE=django.db.backends.postgresql
      - DB_HOST=postgres
      - DB_PORT=5432
    networks:
      - backend

networks:
  backend:
      driver: bridge